bundles/org.openhab.binding.hive/src/main/java/org/openhab/binding/hive/internal/client/feature/HeatingThermostatFeature.java

```diff
@@ -0,0 +1,170 @@
+            final FeatureAttribute<HeatingThermostatOperatingState> operatingState,
+            final SettableFeatureAttribute<Quantity<Temperature>> targetHeatTemperature,
+            final SettableFeatureAttribute<OverrideMode> temporaryOperatingModeOverride
+    ) {
+        Objects.requireNonNull(operatingMode);
+        Objects.requireNonNull(operatingState);
+        Objects.requireNonNull(targetHeatTemperature);
+        Objects.requireNonNull(temporaryOperatingModeOverride);
+
+        this.operatingMode = operatingMode;
+        this.operatingState = operatingState;
+        this.targetHeatTemperature = targetHeatTemperature;
+        this.temporaryOperatingModeOverride = temporaryOperatingModeOverride;
+    }
+
+    public SettableFeatureAttribute<HeatingThermostatOperatingMode> getOperatingMode() {
+        return this.operatingMode;
+    }
+
+    public HeatingThermostatFeature withTargetOperatingMode(final HeatingThermostatOperatingMode targetOperatingMode) {
+        return HeatingThermostatFeature.builder()
+                .from(this)
+                .operatingMode(
+                        DefaultFeatureAttribute.<HeatingThermostatOperatingMode>builder()
+                                .from(this.operatingMode)
+                                .targetValue(targetOperatingMode)
+                                .build()
+                )
+                .build();
+    }
+
+    public FeatureAttribute<HeatingThermostatOperatingState> getOperatingState() {
+        return this.operatingState;
+    }
+
+    public SettableFeatureAttribute<Quantity<Temperature>> getTargetHeatTemperature() {
+        return this.targetHeatTemperature;
+    }
+
+    public HeatingThermostatFeature withTargetTargetHeatTemperature(final Quantity<Temperature> targetTargetHeatTemperature) {
+        return HeatingThermostatFeature.builder()
+                .from(this)
+                .targetHeatTemperature(
+                        DefaultFeatureAttribute.<Quantity<Temperature>>builder()
+                                .from(this.targetHeatTemperature)
+                                .targetValue(targetTargetHeatTemperature)
+                                .build()
+                )
+                .build();
+    }
+
+    public SettableFeatureAttribute<OverrideMode> getTemporaryOperatingModeOverride() {
+        return this.temporaryOperatingModeOverride;
+    }
+
+    public HeatingThermostatFeature withTargetTemporaryOperatingModeOverride(final OverrideMode targetOverrideMode) {
+        return HeatingThermostatFeature.builder()
+                .from(this)
+                .temporaryOperatingModeOverride(
+                        DefaultFeatureAttribute.<OverrideMode>builder()
+                                .from(this.temporaryOperatingModeOverride)
+                                .targetValue(targetOverrideMode)
+                                .build()
+                )
+                .build();
```

Comment:
I'm not sure how I would refactor this. Have you got something in mind?
I have been trying to make everything immutable if I could.
I added these methods because in the `handler.strategy.xxxStrategy` classes I often need to just set the target value for one feature attribute. I though that these methods would make things nicer as a user of my "library".

Reply:
I should have been more specific. I saw that this code, or similar code, is used in several places in this class. My suggestion is to refactor them into a single method.

Reply:
I suppose I could make a method that turns this:
```java
DefaultFeatureAttribute.<OverrideMode>builder()
    .from(this.temporaryOperatingModeOverride)
    .targetValue(targetOverrideMode)
    .build()
```
into this:
```java
this.temporaryOperatingModeOverride.withTarget(targetOverrideMode)
```
Is that what you meant?